main.py

```diff
@@ -153,14 +161,110 @@ async def on_post(self, req: Request, resp: Response):
         resp.data = completion.to_json()
 
 
+class Embeddings:
+    def __init__(self, model_name: str) -> None:
+        self.model_name = model_name
+        self.model = SentenceTransformer(self.model_name)
+        from transformers import AutoTokenizer, AutoModel
+
+        # Load model from HuggingFace Hub
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model)
+        self.model = AutoModel.from_pretrained(self.model)
+
+    def embed_and_get_token_count(self, sentences):
+        # Mean Pooling - Take attention mask into account for correct averaging
+        def mean_pooling(model_output, attention_mask):
+            token_embeddings = model_output[
+                0
+            ]  # First element of model_output contains all token embeddings
+            input_mask_expanded = (
+                attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
+            )
+            return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(
+                input_mask_expanded.sum(1), min=1e-9
+            )
+
+        # Tokenize sentences
+        encoded_input = self.tokenizer(
+            sentences, padding=True, truncation=True, return_tensors='pt'
+        )
+        token_count = encoded_input['attention_mask'].sum(dim=1)
+
+        # Compute token embeddings
+        with torch.no_grad():
+            model_output = self.model(**encoded_input)
+
+        # Perform pooling
+        sentence_embeddings = mean_pooling(
+            model_output, encoded_input['attention_mask']
+        )
+
+        # Normalize embeddings
+        sentence_embeddings = F.normalize(sentence_embeddings, p=2, dim=1)
```

Comment:
Is it copied from the sentence_transformers? If it doesn't expose this interface, you can skip the token count for now and create another issue to track it.